app/src/main/java/org/oppia/android/app/hintsandsolution/HintsAndSolutionDialogFragmentPresenter.kt

```diff
@@ -112,7 +112,7 @@ class HintsAndSolutionDialogFragmentPresenter @Inject constructor(
         bindingAdapter = createRecyclerViewAdapter()
         adapter = bindingAdapter
       }
-      if (viewModel.newAvailableHintIndex.get() != -1) {
+      if (viewModel.newAvailableHintIndex.get() != -2) {
```

Comment:
This seems slightly hacky to me--why is '-2' correct? It seems like a magic number since there isn't any context of where it's coming from.

Reply:
@BenHenning I think this value should be -2 instead of -1 because we are multiplying the newAvailableHintIndex with 2 to account for the divider, so this value will never be -1.

Reply:
That doesn't seem correct @MaskedCarrot. The index isn't the index in the recycler view, it's the hint index within the index list. We shouldn't be conflating the two since one is a domain concept and the other is a view concept.

Reply:
resolving this because changed the approach to handle hints

Reply:
@BenHenning We are multiplying the viewModel.newHintIndex with 2 in line number 93 and then calling the function ` loadHintsAndSolution(state)`  so I think it should be `-2` PTAL.